tokenserver-common/src/error.rs

```diff
@@ -52,31 +73,31 @@ impl TokenserverError {
         }
     }
 
-    pub fn invalid_key_id(description: &'static str) -> Self {
+    pub fn invalid_key_id(description: String) -> Self {
```

Comment:
Alternatively you could take `description: &str` for these to avoid `to_owned()` at the call sites. You'll need to pass `&foo` when you have owned `String`s and it'll add an extra clone, but that's not the common case and IMO the small clone's worth it for convenience (but feel free to disagree and leave as is).

Reply:
(and when you *need* to avoid that extra clone you can take an Into<Cow<&str>> but it's rarely worth that effort)

Reply:
I think I had done it this way to make the clones obvious: in my mind, passing by reference suggests that the function doesn't need ownership, and it may suggest to the caller that the function is less expensive than it really is. But now I'm not so sure 🤔

Reply:
Perhaps having the function take `Into<String>` is a good compromise? It allows flexibility, but it signals to the caller that the function does in fact need ownership

Reply:
I'm ok with how it is now 😄  but good point, `Into<String>` is simpler and works similarly and isn't too much trouble to specify as a generic type.